Storage/SQL/SQLiteHistory.swift

```diff
@@ -75,21 +75,37 @@ func getLocalFrecencySQL() -> String {
 }
 
 fileprivate func escapeFTSSearchString(_ search: String) -> String {
-    // Remove double-quotes and split search string on whitespace.
-    let words = search.replacingOccurrences(of: "\"", with: "").components(separatedBy: .whitespaces)
+    // Remove double-quotes, split search string on whitespace
+    // and remove any empty strings
+    let words = search.replacingOccurrences(of: "\"", with: "").components(separatedBy: .whitespaces).filter({ !$0.isEmpty })
+
+    // If there's only one word, ensure it is longer than 2
+    // characters. Otherwise, form a different type of search
+    // string to attempt to match the start of URLs.
+    guard words.count > 1 else {
+        if let word = words.first {
+            if word.count > 2 {
+                return "\"\(word)*\""
+            } else {
+                return "title: \"^\(word)*\" OR " +
+                    "url: \"^http://\(word)*\" OR " +
+                    "url: \"^https://\(word)*\" OR " +
+                    "url: \"^http://www.\(word)*\" OR " +
+                    "url: \"^https://www.\(word)*\" OR " +
+                    "url: \"^http://m.\(word)*\" OR " +
+                    "url: \"^https://m.\(word)*\""
+            }
+        } else {
+            return ""
+        }
+    }
```

Comment:
I am finding this a bit hard to be sure there isn't a typo somewhere, perhaps something like: 
```
   guard words.count > 1 else {
        guard let word = words.first else {
            return ""
        }

        let charThresholdForSearchAll = 2
        if word.count > charThresholdForSearchAll {
            return "\"\(word)*\""
        } else {
            let titlePrefix = "title: \"^"
            let httpPrefix = "url: \"^http://"
            let httpsPrefix = "url: \"^https://"

             return [titlePrefix,
                     httpPrefix,
                     httpsPrefix,
                     httpPrefix + "www.",
                     httpsPrefix + "www.",
                     httpPrefix + "m.",
                     httpsPrefix + "m."]
                .map { prefix in "\(prefix)\(word)*\""}
                .joined(separator: " OR ")
        }
    }
```

Reply:
I made sure that code generated the same string as the original (I did an `assert()`)